File: web/src/types/brand.ts
// 品牌相关类型定义

export interface Brand {
  id: number;
  tenant_id: number;
  name: string;
  slug: string;
  description: string;
  logo_url: string;
  website: string;
  industry: string;
  founded_year: number;
  headquarters: string;
  status: BrandStatus;
  settings: Record<string, string>;
  created_at: string;
  updated_at: string;
}

export enum BrandStatus {
  Unspecified = 0,
  Active = 1,
  Archived = 2,
  Disabled = 3,
}

export interface BrandMetadata {
  brand_id: number;
  vi_profile: VIProfile;
  tone_profile: ToneProfile;
  audience_profiles: AudienceProfile[];
  competitor_list: CompetitorInfo[];
  brand_values: string[];
  unique_selling_points: string[];
  schema_version: string;
  created_at: string;
  updated_at: string;
}

export interface VIProfile {
  primary_color: string;
  secondary_color: string;
  logo_url: string;
  font_family: string;
  brand_keywords: string[];
  slogan: string;
}

export interface ToneProfile {
  formality: 'formal' | 'casual' | 'technical';
  personality: 'friendly' | 'professional' | 'playful' | 'authoritative';
  avoid_words: string[];
  preferred_phrases: string[];
  style_guide: string;
}

export interface AudienceProfile {
  name: string;
  age_range: string;
  interests: string[];
  pain_points: string[];
  preferred_channels: string[];
  locations: string[];
  languages: string[];
}

export interface CompetitorInfo {
  name: string;
  domain: string;
  description: string;
  strengths: string[];
  weaknesses: string[];
}

export interface GlossaryEntry {
  id: number;
  brand_id: number;
  term: string;
  definition: string;
  category: string;
  aliases: string[];
  context: string;
  is_forbidden: boolean;
  is_preferred: boolean;
  created_at: string;
  updated_at: string;
}

export interface BrandSnapshot {
  id: number;
  brand_id: number;
  version: string;
  snapshot_data: string;
  change_log: string;
  created_by: number;
  created_at: string;
}

export interface KnowledgeEntity {
  id: number;
  brand_id: number;
  name: string;
  type: EntityType;
  description: string;
  properties: Record<string, string>;
  authority_links: string[];
  tags: string[];
  embedding_id: string;
  created_at: string;
  updated_at: string;
}

export enum EntityType {
  Unspecified = 'unspecified',
  Brand = 'brand',
  Product = 'product',
  Person = 'person',
  Org = 'org',
  Event = 'event',
  Concept = 'concept',
  Location = 'location',
  Technology = 'technology',
}

export interface KnowledgeRelation {
  id: number;
  from_entity_id: number;
  to_entity_id: number;
  type: RelationType;
  weight: number;
  description: string;
  properties: Record<string, string>;
  created_at: string;
}

export enum RelationType {
  Unspecified = 'unspecified',
  IsA = 'is_a',
  PartOf = 'part_of',
  RelatedTo = 'related_to',
  CompetesWith = 'competes_with',
  Mentions = 'mentions',
  DependsOn = 'depends_on',
  Owns = 'owns',
  CreatedBy = 'created_by',
}

export interface BrandTrustScore {
  id: number;
  brand_id: number;
  score: number;
  dimensions: {
    search: number;
    social: number;
    compliance: number;
    citation: number;
  };
  factors: string;
  api_request_id: string;
  expires_at: string;
  created_at: string;
}

// 创建品牌请求
export interface CreateBrandRequest {
  name: string;
  slug: string;
  description?: string;
  logo_url?: string;
  website?: string;
  industry?: string;
  founded_year?: number;
  headquarters?: string;
}

// 更新品牌请求
export interface UpdateBrandRequest {
  name?: string;
  description?: string;
  logo_url?: string;
  website?: string;
  industry?: string;
  founded_year?: number;
  headquarters?: string;
  status?: BrandStatus;
  settings?: Record<string, string>;
}

// 创建术语请求
export interface CreateGlossaryEntryRequest {
  term: string;
  definition: string;
  category?: string;
  aliases?: string[];
  context?: string;
  is_forbidden?: boolean;
}

// 批量导入术语请求
export interface BulkImportGlossaryRequest {
  entries: CreateGlossaryEntryRequest[];
  overwrite_existing?: boolean;
}

// 创建知识实体请求
export interface CreateKnowledgeEntityRequest {
  name: string;
  type: EntityType;
  description?: string;
  properties?: Record<string, string>;
  authority_links?: string[];
  tags?: string[];
}

// 创建知识关系请求
export interface CreateKnowledgeRelationRequest {
  from_entity_id: number;
  to_entity_id: number;
  type: RelationType;
  weight?: number;
  description?: string;
  properties?: Record<string, string>;
}

File: service/brand/internal/domain/brand.go
package domain

import (
	"errors"
	"time"
)

// BrandStatus 品牌状态
type BrandStatus int32

const (
	BrandStatusUnspecified BrandStatus = 0
	BrandStatusActive      BrandStatus = 1
	BrandStatusArchived    BrandStatus = 2
	BrandStatusDisabled    BrandStatus = 3
)

// Brand 品牌聚合根
type Brand struct {
	ID           int64
	TenantID     int64
	Name         string
	Slug         string
	Description  string
	LogoURL      string
	Website      string
	Industry     string
	FoundedYear  int32
	Headquarters string
	Status       BrandStatus
	Settings     map[string]string
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

// BrandMetadata 品牌元数据值对象
type BrandMetadata struct {
	BrandID             int64
	VIProfile           VIProfile
	ToneProfile         ToneProfile
	AudienceProfiles    []AudienceProfile
	CompetitorList      []CompetitorInfo
	BrandValues         []string
	UniqueSellingPoints []string
	SchemaVersion       string
	CreatedAt           time.Time
	UpdatedAt           time.Time
}

// VIProfile VI 规范值对象
type VIProfile struct {
	PrimaryColor   string
	SecondaryColor string
	LogoURL        string
	FontFamily     string
	BrandKeywords  []string
	Slogan         string
}

// ToneProfile 语调规范值对象
type ToneProfile struct {
	Formality       string
	Personality     string
	AvoidWords      []string
	PreferredPhrases []string
	StyleGuide      string
}

// AudienceProfile 受众画像值对象
type AudienceProfile struct {
	Name             string
	AgeRange         string
	Interests        []string
	PainPoints       []string
	PreferredChannels []string
	Locations        []string
	Languages        []string
}

// CompetitorInfo 竞品信息值对象
type CompetitorInfo struct {
	Name        string
	Domain      string
	Description string
	Strengths   []string
	Weaknesses  []string
}

// GlossaryEntry 术语条目实体
type GlossaryEntry struct {
	ID          int64
	BrandID     int64
	Term        string
	Definition  string
	Category    string
	Aliases     []string
	Context     string
	IsForbidden bool
	IsPreferred bool
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// BrandSnapshot 品牌快照实体
type BrandSnapshot struct {
	ID           int64
	BrandID      int64
	Version      string
	SnapshotData string
	ChangeLog    string
	CreatedBy    int64
	CreatedAt    time.Time
}

// 领域错误
var (
	ErrBrandNotFound      = errors.New("brand not found")
	ErrBrandSlugExists    = errors.New("brand slug already exists")
	ErrBrandQuotaExceeded = errors.New("brand quota exceeded")
	ErrGlossaryNotFound   = errors.New("glossary entry not found")
	ErrSnapshotNotFound   = errors.New("brand snapshot not found")
)

// Validate 验证品牌实体
func (b *Brand) Validate() error {
	if b.TenantID <= 0 {
		return errors.New("tenant_id is required")
	}
	if b.Name == "" {
		return errors.New("brand name is required")
	}
	if b.Slug == "" {
		return errors.New("brand slug is required")
	}
	return nil
}

// IsActive 检查品牌是否活跃
func (b *Brand) IsActive() bool {
	return b.Status == BrandStatusActive
}

// Validate 验证术语条目
func (g *GlossaryEntry) Validate() error {
	if g.BrandID <= 0 {
		return errors.New("brand_id is required")
	}
	if g.Term == "" {
		return errors.New("term is required")
	}
	if g.Definition == "" {
		return errors.New("definition is required")
	}
	return nil
}
